src/main/java/it/gov/pagopa/payhub/activities/activity/treasury/TreasuryOpiIngestionActivityImpl.java

```diff
@@ -26,50 +28,84 @@
-      return new TreasuryIufResult(Collections.emptyList(), false);
     }
-    return new TreasuryIufResult(iufIuvList, true);
-  }
 
+    private IngestionFlowFileDTO findIngestionFlowFileRecord(Long ingestionFlowFileId) {
+        IngestionFlowFileDTO ingestionFlowFileDTO = ingestionFlowFileDao.findById(ingestionFlowFileId)
+                .orElseThrow(() -> new IngestionFlowFileNotFoundException("Cannot found ingestionFlow having id: " + ingestionFlowFileId));
+        if (!ingestionFlowFileDTO.getFlowFileType().equals(ingestionflowFileType)) {
+            throw new IllegalArgumentException("invalid ingestionFlow file type " + ingestionFlowFileDTO.getFlowFileType());
+        }
+        return ingestionFlowFileDTO;
+    }
 
+    private List<Path> retrieveFiles(IngestionFlowFileDTO ingestionFlowFileDTO) throws IOException {
 
-  private IngestionFlowFileDTO findIngestionFlowFileRecord(Long ingestionFlowFileId) {
-      IngestionFlowFileDTO ingestionFlowFileDTO = ingestionFlowFileDao.findById(ingestionFlowFileId)
-              .orElseThrow(() -> new IngestionFlowFileNotFoundException("Cannot found ingestionFlow having id: " + ingestionFlowFileId));
-      if (!ingestionFlowFileDTO.getFlowFileType().equals(ingestionflowFileType)) {
-        throw new IllegalArgumentException("invalid ingestionFlow file type " + ingestionFlowFileDTO.getFlowFileType());
-      }
-    return ingestionFlowFileDTO;
-  }
+        return ingestionFlowFileRetrieverService
+                .retrieveAndUnzipFile(Path.of(ingestionFlowFileDTO.getFilePath()), ingestionFlowFileDTO.getFileName());
+    }
 
-  private  List<Path> retrieveFiles(IngestionFlowFileDTO ingestionFlowFileDTO) throws IOException {
+    private List<String> parseData(Path ingestionFlowFilePath) {
+        File ingestionFlowFile=ingestionFlowFilePath.toFile();
+
+        it.gov.pagopa.payhub.activities.xsd.treasury.opi14.FlussoGiornaleDiCassa flussoGiornaleDiCassa14 = null;
+        it.gov.pagopa.payhub.activities.xsd.treasury.opi161.FlussoGiornaleDiCassa flussoGiornaleDiCassa161 = null;
+
+        try {
+            flussoGiornaleDiCassa161 = treasuryUnmarshallerService.unmarshalOpi161(ingestionFlowFile);
+            log.debug("file flussoGiornaleDiCassa with Id {} parsed successfully ", flussoGiornaleDiCassa161.getId());
+        } catch (Exception e) {
+            log.error("file flussoGiornaleDiCassa parsing error with opi 1.6.1 format {} ", e.getMessage());
+        }
+        if (flussoGiornaleDiCassa161 == null) {
+            try {
+                flussoGiornaleDiCassa14 = treasuryUnmarshallerService.unmarshalOpi14(ingestionFlowFile);
+                log.debug("file flussoGiornaleDiCassa with Id {} parsed successfully ", flussoGiornaleDiCassa14.getId());
+            } catch (Exception e) {
+                log.error("file flussoGiornaleDiCassa parsing error with opi 1.4 format {} ", e.getMessage());
+            }
+        }
```

Comment:
if we cannot understand the right model without opening the file, it could be ok this approach, but the first exception could not be an error, moreover, the second one yes, it's an error and we should throw an exception: let's introduce a custom Exception to be throw here (eg `TreasuryOpiInvalidFileException`)
```suggestion
            log.info("file flussoGiornaleDiCassa parsing error with opi 1.6.1 format {} ", e.getMessage());
            try {
                flussoGiornaleDiCassa14 = treasuryUnmarshallerService.unmarshalOpi14(ingestionFlowFile);
                log.debug("file flussoGiornaleDiCassa with Id {} parsed successfully ", flussoGiornaleDiCassa14.getId());
            } catch (Exception e) {
                log.info("file flussoGiornaleDiCassa parsing error with opi 1.4 format {} ", e.getMessage());
                throw new TreasuryOpiInvalidFileException("Cannot parse treasury Opi file " + ingestionFlowFile);
            }
        }
```